docs/src/_contribute/cli.md

```diff
@@ -0,0 +1,225 @@
+
+## Getting Set Up
+
+See the prerequisite section for instructions on how to set up a development environment and dependencies.
+
+## CLI Design Guidelines
+
+The `meltano` CLI is the primary interface for interacting with Meltano. It's often the first introduction to meltano
+that a users experiences when browsing the docs or experimenting with the CLI. As such, we aim to provide a clear and
+consistent interface that is easy to use. This guide aims to codify the design guidelines for the CLI so that
+contributors can easily update or extend the CLI with new features, commands, options, and enhancements.
+
+This document is somewhat aspirational and a work in progress. Portions of the CLI may violate the design and style
+guidelines, but we aim to update the CLI overtime to ensure that it is as consistent as possible.
+
+### Consistent definition of Groups, Commands, Sub-commands, Arguments, and Options
+
+Groups, Commands, Sub-commands, Arguments, Options, and Flags often have slightly different meanings for users. So are
+defined as follows for use in `meltano`.
+
+- Group is a group of commands that are related to a specific area of the CLI, typically a specific meltano feature.
+  For example, the `meltano schedule` feature-group contains commands related to managing Meltano schedules.
+- command/sub_command are used to perform specific tasks or used to group a set of command around a specific feature
+  or task. Sub-commands take the form of parameters, but require additional arguments or options themselves.
+- Arguments are positional parameters that are passed to a command. Arguments do not require additional options or
+  arguments - otherwise they would be considered a sub-command.
+- Options (switch, option flags, or flags) are options that alter the behavior (e.g. `--dry-run/--verbose`) or named
+  input options (e.g. `--tasks=`). They come in two forms. A long form: `--option-name`, and a short form: `-o`.
+
+Given a command like `meltano job set JOB_NAME --tasks=[<task>...]`:
+
+- `meltano` is the global click command group for Meltano. This is the main entry point to the CLI. When we refer to global options this is the level we refer to.
+- `job` is the click command group for the `job` command. This is the main command group for the `job` command. This is typically the top level for a Meltano feature.
+- `set` is a `sub-command` of the `job` command.
+- `JOB_NAME` is an `argument` of the `set` sub-command.
+- `--tasks` is named `option` of the `set` sub-command.
+
+For a technical explanation of how commands and groups work see https://click.palletsprojects.com/en/8.0.x/commands/
+
+### Use of short and long options
+
+The options you expect to be used most frequently should always include both the long AND short variation of the option.
+For example `--help` and `-h`, `--tasks` and `-t`, `--force` and `-f`.
+
+
+### Desired verb/command linkage and structure
+
```

Comment:
@tayloramurphy @aaronsteers In the past we've had a lot back and forth on how to structure the cli components and it sounds like y'all had differing philosophies on this perhaps. So to get us started, I only briefly described what we did for `job` and what we refactored `schedule` too. 

If this is the structure we want to maintain moving forward I can send along an update with more perscriptive guidance and examples.

Reply:
@pandemicsyn this is a great PR. I love clear guidance docs like this. I'm quite happy with the structure we have for `jobs` and `schedule` and I'm happy to move forward with that. 

The only note I had is to add guidance for when (or when not) to use angle brackets `<>` in the docs and/or help text.

Reply:
@tayloramurphy Yea, i was tempted to explicitly say to never use. Buts its kinda handy and pretty common. The problem is, we use `<stuff>` interchangeably as a placeholder/var substitution for both required AND optional items, so it makes it a bit confusing. What about adding "placeholders" section that clarifies use of angle brackets, as well as when to use generic names like `tap`, `target` vs `tap-gitlab` in examples?

Reply:
@pandemicsyn "placeholders" makes sense 👍 I agree we have to have something b/c I'm not sure how we're going to get away from having `<>` in the docs at all 😅

Reply:
👍 added a section.

tl;dr better for docs than CLI help out. If you use the angle bracket pattern you should still follow the rest of the rules e.g. using `[<MAPPING1> ...]` to indicate whether its required/optional. 

@tayloramurphy also i said to prefer Tap/Target over Extractor/Loader. Is the inverse better?

Reply:
@pandemicsyn tap/target over extractor/loader I think is fine - that's primarily what we're using in the docs currently, though usually scoped to the name (tap_name / target_name). 

Otherwise the addition looks good with the 2 minor comments I had.